File: Week1/pset1/mario.c
/*
*AUTHOR: Branko Stankovic
contact: branko.stankovic@gmail.com
webpage: http://meanwebdeveloper.wordpress.com
*DATE CREATED: December 22nd 2016.
*/
#include <stdio.h>
#include <cs50.h>

int main(void) 
{
    int height;
    
    // takes the user input
    do
    {
    printf("Height?");
    height = GetInt();
    }
    while (!(height>=0 && height<24));
    
    int spaces = height-1;
    int hashes = 2;
    
    // draws the pyramid
    for (int i=0; i<height; i++) 
    {
        for (int k=spaces; k>0; k--) 
        {
            printf(" ");
        }
        spaces--;
        for (int j=hashes; j>0; j--) 
        {
            printf("#");
        }
        printf("\n");
        hashes++;
    }
}

File: pset6/mario.py
# branko stankovic
# cs 50 edX course
# https://branko-stankovic.github.io/

# cs50 library
import cs50

# forever loop, until it meets my condition
while True:
    # getting user input
    print("Height? ", end="")
    height = cs50.get_int()
    
    if height > 0 and height < 24:
        break

# setup for the top row of pyramid
hashes = 1
spaces = height - 1

# now building the pyramid!
# it is i stories high
for i in range(height):
    # making some space for mario to jump
    print(" " * spaces, end="")
    # drawing the actual blocks of pyramid
    print("#" * hashes, end="")
    # space between the two halfs of the pyramid    
    print("  ", end="")
    # and the second part of pyramid now
    print("#" * hashes, end="")
    # at each iteration, values change for the next row of pyramid
    hashes = hashes + 1  
    spaces = spaces - 1
    # end of line hack
    print(" ")